editor/store/selectors.js

```diff
@@ -238,10 +238,11 @@ export function isCurrentPostPending( state ) {
  * @return {boolean} Whether the post has been published.
  */
 export function isCurrentPostPublished( state ) {
-	const post = getCurrentPost( state );
+	const status = getEditedPostAttribute( state, 'status' );
```

Comment:
II also believe we wanted to get the current post's status and date here and not the potentially updated ones.